internal/engine/filter/filter.go

```diff
@@ -67,6 +68,67 @@ func (f Filter) String() string {
 	return w.String()
 }
 
+func indent(query string, level int) string {
+	return strings.TrimRight(indentInternal(query, level), "\n ")
+}
+
+func indentInternal(query string, level int) string {
+	var out bytes.Buffer
+	firstChar := true
+	sawColon := true
+	inQuotes := false
```

Comment:
I wonder if this should be an enum `state` instead. Is there a moment where colon and inQuotes should both be true?